src/main/java/com/gaggle/contact/controllers/ContactController.java

```diff
@@ -44,6 +46,20 @@
                 .orElseThrow(() -> new ContactNotFoundException(name));
     }
 
+    @GetMapping("/query")
+    public List<Contact> getByEmail(@RequestParam final String email) {
+        if (Objects.isNull(email) || !email.contains("@")) {
+            throw new RuntimeException("email is no good, try again silly");
+        }
+
+        Optional<List<Contact>> contacts = repository.findByEmailContaining(email);
```

Comment:
What DB are we using? Some can be case sensitive and some aren't (SQL Server 2012 vs PostGRES), per requirement 3 text case should be ignored when searching by email.